File: langs/typescript/src/localization/nullLocalizer.ts
/**
 * NullLocalizer — null-object variant per ADR-0017.
 */
import type { ILocalizer } from "./localizer.js";

export class NullLocalizer implements ILocalizer {
  static readonly INSTANCE: NullLocalizer = new NullLocalizer();

  private constructor() {}

  localize(key: string, _args?: readonly unknown[]): string {
    return key;
  }
}

File: langs/swift/Sources/VMx/Localization/NullLocalizer.swift
/// Null variant: returns the key verbatim (LOC-002), ignoring any args.
public final class NullLocalizer: Localizer, Sendable {
    public static let INSTANCE = NullLocalizer()
    public init() {}
    public func localize(_ key: String, _ args: [Any]?) -> String { key }
}
